core/primitives-core/src/runtime/fees.rs

```diff
@@ -217,8 +217,17 @@ pub fn transfer_exec_fee(
         (_, AccountType::NamedAccount) => transfer_fee,
         // No account will be created, just a regular transfer.
         (false, _) => transfer_fee,
-        // Currently, no account is created on transfer to ETH-implicit account, just a regular transfer.
-        (true, AccountType::EthImplicitAccount) => transfer_fee,
+        // Extra fee for the CreateAccount.
+        (true, AccountType::EthImplicitAccount) => {
+            #[cfg(feature = "protocol_feature_eth_implicit")]
+            {
+                transfer_fee + cfg.fee(ActionCosts::create_account).exec_fee()
+            }
+            #[cfg(not(feature = "protocol_feature_eth_implicit"))]
+            {
+                transfer_fee
+            }
+        }
```

Comment:
This logic needs to be dependent on the protocol version. Please use of checked_feature! macro. Same for any other similar code. 

If possible I would remove the rust feature conditionality and use the dynamic checks based on the protocol version. Just to make sure it's the current agreement let's get an agreement on [zulip](https://near.zulipchat.com/#narrow/stream/295302-general/topic/gating.20new.20features)

Reply:
@wacban How about now?